src/components/NanoContract/EditAddressModal.component.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { t } from 'ttag';
+import { COLORS } from '../../styles/themes';
+import { CircleInfoIcon } from '../Icons/CircleInfo.icon';
+import { ModalBase } from '../ModalBase';
+import { TextValue } from '../TextValue';
+
+/**
+ * Use this modal to edit an address.
+ *
+ * @param {Object} props
+ * @param {string} props.item It refers to the address selected
+ * @param {boolean} props.show It determines if modal must show or hide
+ * @param {(address:string) => {}} props.onAddressChange
+ * Function called when an address is selected
+ * @param {() => {}} props.onDismiss
+ * Function called to dismiss the modal on user's interaction
+ *
+ * @example
+ * <EditAddressModal
+ *   show={showEditAddressModal}
+ *   item={selectedAddress}
+ *   onDismiss={onDismissEditAddressModal}
+ *   onAddressChange={handleAddressChange}
+ * />
+ */
+export const EditAddressModal = ({ item, show, onAddressChange, onDismiss }) => (
+  <ModalBase show={show} onDismiss={onDismiss}>
+    <ModalBase.Title>{t`New Nano Contract Address`}</ModalBase.Title>
+    <ModalBase.Body style={styles.body}>
+      <View style={styles.infoContainer}>
+        <View style={styles.infoIcon}>
+          <CircleInfoIcon size={20} color='hsla(203, 100%, 25%, 1)' />
+        </View>
+        <View style={styles.infoContent}>
+          <Text style={styles.text}>
+            {t`This address signs any transaction you create with Nano Contracts method. Switching to a new one means`}
+            <Text style={styles.bold}>
+              {' '}{t`all future transactions will use this address.`}
+            </Text>
+          </Text>
+        </View>
+      </View>
+      <View style={styles.selectionContainer}>
+        <Text style={[styles.text, styles.bold, styles.pd8]}>{t`Selected Information`}</Text>
+        <FieldContainer>
+          <Text>
+            <TextValue bold>{t`Address`}</TextValue>
+            {/* the unicode character u00A0 means no-break space. */}
+            <TextValue>{(`: ${item.address}`).replaceAll(' ', '\u00A0')}</TextValue>
```

Comment:
This technique results in oneline text:

<img src="https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/5a3dd697-21b1-4eca-bb6f-c44ada72e28f" height="600" />

<img src="https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/147ee2a9-60ef-4ecb-8e97-33da3c7c16ca" height="600" />

Reply:
Yeah, it is this one you pointed out.

Reply:
I've changed a little bit this implementation. Now I'm using the character directly on the space spot. No need to use `relplaceAll` indeed.